Modules/Utils/Sources/PocketCastsUtils/General/KPMSearch.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+
+// An implementation of the Knuth-Morris-Pratt algorithm
+public class KMPSearch {
+    private var pattern: [Character]
+    private var lps: [Int]
+
+    public init(pattern: String) {
+        self.pattern = Array(pattern.lowercased())
+        self.lps = Array(repeating: 0, count: pattern.count)
+        computeLPSArray()
+    }
+
+    private func computeLPSArray() {
+        guard lps.count > 0 else {
+            return
+        }
+
+        var length = 0 // length of the previous longest prefix suffix
+        lps[0] = 0 // lps[0] is always 0
+        var i = 1
+
+        while i < pattern.count {
+            if pattern[i] == pattern[length] {
+                length += 1
+                lps[i] = length
+                i += 1
+            } else {
+                if length != 0 {
+                    length = lps[length - 1]
+                } else {
+                    lps[i] = 0
+                    i += 1
+                }
+            }
+        }
+    }
+
+    public func search(in text: String) -> [Int] {
+        guard pattern.count > 0 else {
```

Comment:
Same here about the `guard`

Reply:
Done in 1dcaa25016d61417b07fd8165d3fab0a4434ede5